File: Two Pointer/validPlaindrome.go
package TwoPointer

import "unicode"


func isValidPlaindroem(s string) bool {
	txt := []rune(s)
	start := 0
	end := len(txt)-1

	for start < end {
		if !unicode.IsLetter(txt[start])&& !unicode.IsNumber(txt[start]){
			start++
			continue
		}
		if !unicode.IsLetter(txt[end]) && !unicode.IsNumber(txt[end]){
			end--
			continue
		}
		if unicode.ToLower(txt[start]) != unicode.ToLower(txt[end]) {
			return false
		}
		start++
		end--

	}
	return true

}

File: Two Pointer/validPlaindrome.py

def is_valid_plaindrome(s):
    txt = ''.join(c.lower() for c in s if c.isalnum())
    start = 0
    end = len(txt) -1
    
    while start < end:
        if txt[start] != txt[end]:
            return False
        start += 1
        end -= 1
    return True
